src/tree/ResolvableTreeItemBase.ts

```diff
@@ -61,9 +64,10 @@ export abstract class ResolvableTreeItemBase extends AzExtParentTreeItem impleme
                 }
 
                 this.resolveResult.contextValuesToAdd?.forEach(cv => this.contextValues.add(cv));
-            });
-
-            // It is not needed to refresh at this point, because `runWithTemporaryDescription` already does that
+            } finally {
+                this._tempDescription = undefined;
+                this.treeDataProvider.refreshUIOnly(this.parent);
```

Comment:
Why is it refreshing the parent?

Reply:
The two spinners that appear are:
1. VS Code spinner that only appears when `getChildren` is called (`loadMoreChildrenImpl`)
2. the tree item icon which we can set to `loading~spin`.

To fix this, we just need to figure out why the VS Code spinner is appearing, aka why is VS Code trying to load the children when these items aren't expanded?

I discovered that calling `refreshUIOnly(this)` where `this` is the child being resolved resulted in a call to `loadMoreChildrenImpl`, while calling `refreshUIOnly(this.parent)` doesn't. 

Did some searching and it's actually a VS Code bug (https://github.com/microsoft/vscode/issues/121567).

---

Side note:

Fixing this seems to speed up the resolving quite a bit, I think this bug might have been (and probably still is in other places) causing a lot of unnecessary calls to `loadMoreChildrenImpl`. We should ask VS Code to investigate fixing this.